eox_tenant/signals.py

```diff
@@ -0,0 +1,180 @@
+# -*- coding: utf-8 -*-
+"""
+Eox Tenant Signals
+==================
+This module makes it possible to override the settings object for a tenant.
+
+How to debug in devstack
+========================
+
+Add the following lines to the lms.envs.devstack_docker module
+
+LOGGING['handlers']['console']['level'] = 'DEBUG'
+LOGGING['loggers']['']['level'] = 'INFO'
+LOGGING['loggers']['eox_tenant'] = {
+    'handlers': ['console'],
+    'propagate': False,
+    'level': 'DEBUG'
+}
+
+"""
+import logging
+import os
+
+from datetime import datetime
+
+from celery.signals import after_task_publish, before_task_publish
+from django.conf import settings as base_settings
+from django.core import signals
+
+from eox_tenant.backends.database import EdunextCompatibleDatabaseMicrositeBackend
+
+
+LOG = logging.getLogger(__name__)
+
+# Read at the beginning so this can not be modified by the tenant configs
+EOX_MAX_CONFIG_OVERRIDE_SECONDS = getattr(base_settings, "EOX_MAX_CONFIG_OVERRIDE_SECONDS", 300)
+
+
+def _update_settings(domain):
+    """
+    Perform the override procedure on the settings object
+    """
+    config, tenant_key = _get_tenant_config(domain)
+
+    if not config.get("EDNX_USE_SIGNAL"):
+        LOG.info("Site %s, does not use Signals", domain)
+        return
+
+    base_settings.__setattr__("EDNX_TENANT_KEY", tenant_key)
+    base_settings.__setattr__("EDNX_TENANT_DOMAIN", domain)
+    base_settings.__setattr__("EDNX_TENANT_SETUP_TIME", datetime.now())
+
+    LOG.debug("PID: %s CONFIGURING THE SETTINGS OBJECT | %s", os.getpid(), tenant_key)
+    for key, value in config.iteritems():
+        if isinstance(value, dict):
+            merged = base_settings.__getattr__(key).copy()
```

Comment:
why its using internal method `__getattr___` instead of the global function? (e.g. get_attr(base_settings, key))